opencart 1.5.6.4/upload/admin/controller/module/smaily_for_opencart.php

```diff
@@ -99,6 +101,30 @@ public function index() {
         $this->data['rss_feed_title'] = $this->language->get('rss_feed_title');
         $this->data['rss_feed_text'] = $this->language->get('rss_feed_text');
         $this->data['rss_feed_url'] = $this->config->get('config_url') . 'index.php?route=smailyforopencart/rss';
+
+        // Layout text and buttons
+        $this->data['entry_title_layout'] = $this->language->get('entry_title_layout');
+        $this->data['entry_title_position'] = $this->language->get('entry_title_position');
+        $this->data['entry_title_status'] = $this->language->get('entry_title_status');
+        $this->data['entry_title_sort_order'] = $this->language->get('entry_title_sort_order');
+        $this->data['entry_select_top'] = $this->language->get('entry_select_top');
+        $this->data['entry_select_bottom'] = $this->language->get('entry_select_bottom');
+        $this->data['entry_select_left'] = $this->language->get('entry_select_left');
+        $this->data['entry_select_right'] = $this->language->get('entry_select_right');
+        $this->data['entry_button_module'] = $this->language->get('entry_button_module');
+        $this->data['button_remove'] = $this->language->get('button_remove');
+
+        // Layout module
+        $this->load->model('design/layout');
+        $this->data['layouts'] = $this->model_design_layout->getLayouts();
+        $this->data['modules'] = array();
```

Comment:
What are `modules` in the context of Smaily for OpenCart module?

Reply:
I think module settings, should I rename it to it for clarity?
This a comment for that block from a newer OC version
`This block parses the Module Settings such as Layout, Position,Status & Order Status to the view* `

Reply:
But (for example) does the `$this->data['modules']` array content correspond to module settings?

To me "module" is an OpenCart module (like Smaily for OpenCart), and "module settings" every configuration option for that module. And the value assigned to `$this->data['modules']` contains only settings for the OPT-IN form.

Reply:
You are not using `modules` to reference `optin_form_settings` any more, so change `$this->data['modules']` to `$this->data['optin_form_settings']`.